framework/web/Request.php

```diff
@@ -842,8 +950,8 @@ public function setUrl($value)
      */
     protected function resolveRequestUri()
     {
-        if (isset($_SERVER['HTTP_X_REWRITE_URL'])) { // IIS
-            $requestUri = $_SERVER['HTTP_X_REWRITE_URL'];
+        if ($this->headers->has('X-Rewrite-Url')) { // IIS
```

Comment:
Should this not be a secure header as well?
Many security tools / proxies validate the URL using white / blacklists. (For example only allow `/admin` from the intranet)
This will allow an attack to bypass it.